wagtail/project_template/Dockerfile

```diff
@@ -1,27 +1,60 @@
-# Use an official Python runtime as a parent image
-FROM python:3.7
-LABEL maintainer="hello@wagtail.io"
+# Use an official Python runtime based on Debian 10 "buster" as a parent image.
+FROM python:3.8.1-slim-buster
 
-# Set environment varibles
-ENV PYTHONUNBUFFERED 1
-ENV DJANGO_ENV dev
+# Add user that will be used in the container.
+RUN useradd wagtail
 
-COPY ./requirements.txt /code/requirements.txt
-RUN pip install --upgrade pip
-# Install any needed packages specified in requirements.txt
-RUN pip install -r /code/requirements.txt
-RUN pip install gunicorn
+# Port used by this container to serve HTTP.
+EXPOSE 8000
 
-# Copy the current directory contents into the container at /code/
-COPY . /code/
-# Set the working directory to /code/
-WORKDIR /code/
+# Set environment variables.
+# 1. Force Python stdout and stderr streams to be unbuffered.
+# 2. Set PORT variable that is used by Gunicorn. This should match "EXPOSE"
+#    command.
+ENV PYTHONUNBUFFERED=1 \
```

Comment:
Had trouble getting this working without `PYTHONPATH`. Also, is it worth setting `DJANGO_SETTINGS_MODULE` here?

Reply:
> Had trouble getting this working without PYTHONPATH.

The only downside should be that you are unable to use `django-admin` commands and need to run `manage.py` commands from the `/app` directory. Or are you talking about some other issues? I personally think setting the `PYTHONPATH` variable is a good idea so people can use `django-admin`, but is that too opinionated and adds too much bloat to the file? I can't see the downside of adding this.

> DJANGO_SETTINGS_MODULE
I think we should use the default one (set in `wsgi.py` or `manage.py`) to not bring too much opinion here. I think we should look at re-factor of the settings to make it one file only as a separate pull request. This is not an avenue for this IMO.

Reply:
@zerolab tried this but didn't hit the issue I had so this was probably to do with the project I was trying it out on.